src/realm/object-store/sync/sync_session.cpp

```diff
@@ -846,12 +846,23 @@ void SyncSession::create_sync_session()
     session_config.proxy_config = sync_config.proxy_config;
     session_config.simulate_integration_error = sync_config.simulate_integration_error;
     session_config.flx_bootstrap_batch_size_bytes = sync_config.flx_bootstrap_batch_size_bytes;
-    if (sync_config.on_sync_client_event_hook) {
-        session_config.on_sync_client_event_hook = [hook = sync_config.on_sync_client_event_hook,
-                                                    anchor = weak_from_this()](const SyncClientHookData& data) {
-            return hook(anchor, data);
-        };
-    }
+    session_config.on_sync_client_event_hook =
```

Comment:
Or you could move the call to `create_subscriptions()` into `process_flx_bootstrap_message()` where the debug hook function is called: https://github.com/realm/realm-core/blob/master/src/realm/sync/client.cpp#L821
(although you we need to grab a reference to the SyncSession from there...) Or perhaps you could use `on_flx_sync_progress()` when the bootstrap is complete.

Reply:
I am looking to change this bit. The rest of the code is good to go.

Reply:
What if instead of hooking this behavior from object-store, we just allowed the sync::Session to have a ref to the MigrationStore and then we could do this directly in the integrate_server_changesets function rather than via a debug hook. That'd also mean that the debug hook can remain a debug hook that is mostly free if a hook isn't defined.

Reply:
great suggestion! i wasn't happy with this either..